system-test/install.ts

```diff
@@ -0,0 +1,37 @@
+import * as execa from 'execa';
+import * as mv from 'mv';
+import {ncp} from 'ncp';
+import * as tmp from 'tmp';
+import {promisify} from 'util';
+
+const keep = false;
+const mvp = promisify(mv);
+const ncpp = promisify(ncp);
+const stagingDir = tmp.dirSync({keep, unsafeCleanup: true});
+const stagingPath = stagingDir.name;
+const pkg = require('../../package.json');
+
+describe('📦 pack and install', () => {
+  it('should be able use the library from a TypeScript application', async function() {
+    this.timeout(300000);
```

Comment:
I appreciate this system-test, are these automatically generated?

Reply:
The one downside to this is that in pulls in a bunch of dependencies (tmp, execa, mv, ncp, and their corresponding types packages).  It kinda adds a lot to the renovate tax as semver majors come out. 

An attempt was made last year to encapsulate this into a module:
https://github.com/google/pack-n-play

I've been poking at this thing to breath live back into it.  What do folks think about nursing that module to health and using it instead of copying all this into each repository?

Reply:
Sounds like a great idea. One thing `pack-n-play` does not check as of today is `npm pack` (it just makes a tarball of the package being tested), I submitted https://github.com/google/pack-n-play/issues/56 for that. I think @xiaozhenliu-gg5 and I can work on that module and make it perfect - but I don't want that work to block our TypeScript releases here. I submitted https://github.com/googleapis/gapic-generator-typescript/issues/129 for the generator and we'll eventually (soon) replace this hand-made system test with the one using `pack-n-play`, but for now I suggest going with this one.

Reply:
Sounds great!,  `pack-n-play`  now is supported to replace system-test: https://github.com/googleapis/gapic-generator-typescript/pull/132/